### PR TITLE
Chore/move renderer globals - Round 2

### PR DESCRIPTION
**Pull request by alex-courtis**
Merge #1279 first.

Simplifies a lot.

help doc has some bad formatting, mixing tabs and spaces. I'll fix the whole file later.

### CODE REVIEW

File: lua/nvim-tree/renderer/icon-config.lua [hunk 1]
```diff
@@ -1,61 +0,0 @@
-local M = {}
```
Comment (alex-courtis) [inline lua/nvim-tree/renderer/icon-config.lua:1]: goodbye

File: lua/nvim-tree/renderer/components/icons.lua [hunk 2]
```diff
@@ -71,23 +69,23 @@ local function config_file_icon()
...
+function M.reset_config()
+  config_symlinks()
   config_file_icon()
   config_folder_icon()
 end
 
+function M.setup(opts)
+  M.config = opts.renderer.icons
+
+  M.devicons = pcall(require, "nvim-web-devicons") and require "nvim-web-devicons"
```
Comment (alex-courtis) [inline lua/nvim-tree/renderer/components/icons.lua:88]: Only needed once.

File: lua/nvim-tree/renderer/components/git.lua [hunk 3]
```diff
@@ -146,21 +143,17 @@ local function get_highlight_(node)
   return git_hl[git_status]
 end
 
-M.get_icons = nil_
-M.get_highlight = nil_
+function M.setup(opts)
```
Comment (alex-courtis) [inline lua/nvim-tree/renderer/components/git.lua:146]: Reload was unnecessary; these can be set once.

File: lua/nvim-tree/colors.lua [hunk 4]
```diff
@@ -84,9 +83,6 @@ local function get_links()
 end
 
 function M.setup()
-  if icons.get_config().show_file_icon and icons.get_config().has_devicons then
-    require("nvim-web-devicons").setup()
```
Comment (alex-courtis) [inline lua/nvim-tree/colors.lua:88]: Moved to nvim-tree.setup

File: doc/nvim-tree-lua.txt [hunk 5]
```diff
@@ -131,6 +131,35 @@ Values may be functions. Warning: this may result in unexpected behaviour.
           git_placement = "before",
           padding = " ",
           symlink_arrow = " ➛ ",
+          show = {
+            file = true,
```
Comment (alex-courtis) [inline doc/nvim-tree-lua.txt:135]: Plurals were removed e.g. files -> file, as they were inconsistent.

File: doc/nvim-tree-lua.txt [hunk 6]
```diff
@@ -131,6 +131,35 @@ Values may be functions. Warning: this may result in unexpected behaviour.
...
+            folder = true,
+            folder_arrow = true,
+            git = true,
+          },
+          symbols = {
```
Comment (alex-courtis) [inline doc/nvim-tree-lua.txt:140]: `renderer.icons.symbols` was the best name I could think of. Ideas welcome.

Comment (kyazdani42) [reply doc/nvim-tree-lua.txt:140]: sounds good to me :)

Comment (alex-courtis) [reply doc/nvim-tree-lua.txt]: Sorry... changed my mind... `glyphs` now.

Matches https://www.nerdfonts.com/ wording.

**Pull request merged**